### PR TITLE
Refactor uncommanded descend Quad-Chute

### PR DESCRIPTION
**Pull request by RomanBapst**
### Solved Problem
The current implementation of the uncommanded descend quadchute revealed two issues:
1) It was possible for a vehicle to enter a limit cycle in which it would stall, loose a lot of altitude, recover, slightly climb up, stall, ...
This behavior has been observed on a vehicle that suffered an airspeed sensor failure and was flying with a too low trim throttle value. In the phases during which the vehicle climbed again after recovering from the stall, the quadchute height rate error integral (current implementation) was reset to 0 and therefore quadchute never triggered, although the vehicle lost close to 100m altitude.

2) Users found it difficult to tune the quadchute as there was no one to one relation between the parameter value and the actual altitude error at which quadchute would trigger.

### Solution
I am proposing a new solution which uses an "altitude error" criteria to trigger quadchute. This is similar to what was implemented in the past with an important difference. The altitude reference used to compute the altitude error is set to the maximum altitude the vehicle has achieved from the time since it has flown BELOW the current altitude setpoint. Additionally, the altitude reference is constrained not to be larger than the current altitude setpoint (in case the setpoint decreases). We use the value of the TECS reference model as our current altitude setpoint.

This method allows quadchute to only be triggered if the vehicle is actually losing altitude. Quadchute will not trigger if the altitude setpoint is increasing faster than what the vehicle can follow, as long as the vehicle is not descending, no quadchute will be triggered.

### Changelog Entry
For release notes:
Refactor uncommanded descend quad-chute to use deterministic altitude error threshold.
```
Feature/Bugfix XYZ
New parameter: VT_QC_ALT_LOSS
Documentation: Need to clarfiy page ... / done, read docs.px4.io/...
```

### Test coverage
Simulation tests have been done.

### Context
Related links, screenshot before/after, video

### CODE REVIEW

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 1]
```diff
@@ -197,12 +197,12 @@ PARAM_DEFINE_FLOAT(VT_FW_MIN_ALT, 0.0f);
 /**
  * Quad-chute uncommanded descent threshold
  *
- * Threshold for integrated height rate error to trigger a uncommanded-descent quad-chute.
- * Only checked in altitude-controlled fixed-wing flight.
- * Additional conditions that have to be met for uncommanded descent detection are a positive (climbing) height
- * rate setpoint and a negative (sinking) current height rate estimate.
+ * Altitude error threshold for quad-chute triggering during fixed-wing flight.
```
Comment (sfuhrer) [inline src/modules/vtol_att_control/vtol_att_control_params.c:200]: Put back the comment "Only checked in altitude-controlled fixed-wing flight."?

**Review by sfuhrer — COMMENTED** [review]
I see no loop whole in the logic, and definitely makes it easier to understand/tune. I have two comments but otherwise good to go and get testing done with it!

**Comment by RomanBapst** [comment]
@ThomasRigi Do you have some comments on this?

**Comment by sfuhrer** [comment]
> @ThomasRigi Do you have some comments on this?

@RomanBapst I think he's currently (next months) not around.

**Comment by sfuhrer** [comment]
Flight tested: 
image

Worded as expected, no false positives and correct trigger eg here when set to 10m error.

**Comment by RomanBapst** [comment]
@sfuhrer Thanks for testing!

**Pull request merged**